src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift

```diff
@@ -428,15 +428,18 @@ import ActiveLabel
 
     private func openProject(withLocalName localProjectName: String?) {
         showLoadingView()
-        CATransaction.flush()
 
-        guard let selectedProject = Project.init(loadingInfo: ProjectLoadingInfo.init(forProjectWithName: localProjectName, projectID: project.projectID)) else {
-            hideLoadingView()
-            Util.alert(text: kLocalizedUnableToLoadProject)
-            return
-        }
+        DispatchQueue.global(qos: .userInitiated).async {
+            guard let selectedProject = Project.init(loadingInfo: ProjectLoadingInfo.init(forProjectWithName: localProjectName, projectID: self.project.projectID)) else {
+                self.hideLoadingView()
+                Util.alert(text: kLocalizedUnableToLoadProject)
```

Comment:
We should probably return to main thread for these UI updates too

Reply:
Could not agree more. Thanks for the detailed review. Changes have been pushed.